python/python/tests/test_async_client.py

```diff
@@ -6351,6 +6351,185 @@ async def test_xautoclaim_edge_cases_and_failures(
                 string_key, group_name, consumer, 0, stream_id0_0
             )
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_xinfo_groups_xinfo_consumers(
+        self, redis_client: TGlideClient, protocol
+    ):
+        key = get_random_string(10)
+        group_name1 = get_random_string(10)
+        group_name2 = get_random_string(10)
+        consumer1 = get_random_string(10)
+        consumer2 = get_random_string(10)
+        stream_id0_0 = "0-0"
+        stream_id1_0 = "1-0"
+        stream_id1_1 = "1-1"
+        stream_id1_2 = "1-2"
+        stream_id1_3 = "1-3"
+
+        # setup: add 3 entries to stream, create consumer group and consumer1, read 1 entry from stream with consumer1
+        assert (
+            await redis_client.xadd(
+                key, [("f1", "v1"), ("f2", "v2")], StreamAddOptions(stream_id1_0)
+            )
+            == stream_id1_0.encode()
+        )
+        assert (
+            await redis_client.xadd(key, [("f3", "v3")], StreamAddOptions(stream_id1_1))
+            == stream_id1_1.encode()
+        )
+        assert (
+            await redis_client.xadd(key, [("f4", "v4")], StreamAddOptions(stream_id1_2))
+            == stream_id1_2.encode()
+        )
+        assert await redis_client.xgroup_create(key, group_name1, stream_id0_0) == OK
+        assert await redis_client.xreadgroup(
+            {key: ">"}, group_name1, consumer1, StreamReadGroupOptions(count=1)
+        ) == {key.encode(): {stream_id1_0.encode(): [[b"f1", b"v1"], [b"f2", b"v2"]]}}
+
+        # sleep to ensure the idle time value and inactive time value returned by xinfo_consumers is > 0
+        time.sleep(2)
+        consumers_result = await redis_client.xinfo_consumers(key, group_name1)
+        assert len(consumers_result) == 1
+        consumer1_info = consumers_result[0]
+        assert consumer1_info.get(b"name") == consumer1.encode()
+        assert consumer1_info.get(b"pending") == 1
+        assert cast(int, consumer1_info.get(b"idle")) > 0
+        if not await check_if_server_version_lt(redis_client, "7.2.0"):
```

Comment:
Should we test that `idle` has different meanings before and after 7.2.0? From docs:

> Note that before Redis 7.2.0, idle used to denote the time passed since last successful interaction. In 7.2.0, inactive was added and idle was changed to denote the time passed since last attempted interaction.

Reply:
Personally I don't think its necessary, but if someone else disagrees let me know. I think we've tested the API functionality well and adding that extra test would be testing Redis functionality more than the API (although I admit the line between the two can be hazy sometimes)